src/dynamic_diagnostic.rs

```diff
@@ -0,0 +1,72 @@
+use std::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+
+use crate::Diagnostic;
+
+/// Diagnostic that can be created at runtime.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DynamicDiagnostic {
+    /// Displayed diagnostic description
+    pub description: String,
+    /// Unique diagnostic code to look up more information
+    /// about this Diagnostic. Ideally also globally unique, and documented
+    /// in the toplevel crate's documentation for easy searching.
+    /// Rust path format (`foo::bar::baz`) is recommended, but more classic
+    /// codes like `E0123` will work just fine.
+    pub code: Option<String>,
+}
```

Comment:
Are we gonna have the other fields here too?

Reply:
Of course!

Reply:
Almost everything from `Diagnostic` is added. I don't know wether `MietteDiagnostic` should (and can?) contain `diagnostic_source`, `related` and `source_code`.

Reply:
These are dynamic over `Diagnostic` and `SourceCode` , that's why they would prevent `MietteDiagnostic` from implementing `PatialEq`, `Eq`, `Clone`, etc.

Reply:
And this won't fit in my needs for example (I need to serialize/deserialize diagnostics with json and compare them for equality)